examples/tutorials/tutorial2-python-vector.ipynb

```diff
@@ -0,0 +1,297 @@
+    "\n",
+    "While libCEED's focus is on high-order finite/spectral element method implementations, the approach is mostly algebraic and thus applicable to other discretizations in factored form, as explained in the [user manual](https://libceed.readthedocs.io/) and [Doxygen documentation](https://codedocs.xyz/CEED/libCEED/)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setting up libCEED for Python\n",
+    "\n",
+    "Install libCEED for Python by running"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "! python -m pip install libceed"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## CeedVector\n",
+    "\n",
+    "Here we show some basic examples to illustrate the `CeedVector` class. In libCEED, CeedVectors constitute the main data structure and serve as input/output for `CeedOperator`. \n",
+    "\n",
+    "We illustrate the simple creation of a `CeedVector`, how to specify its size, and how to read or manipulate its data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import libceed\n",
+    "\n",
+    "ceed = libceed.Ceed()\n",
+    "\n",
+    "n = 10\n",
+    "x = ceed.Vector(n)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can visualize the (unset) CeedVector by typing"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(x)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The size of the `CeedVector` can also be specified as "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "x = ceed.Vector(size=10)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "* In the following example, we associate the data stored in a `CeedVector` with a `numpy.array` and use it to set and read the CeedVector's data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[1. 2. 3.]\n"
+     ]
+    }
+   ],
+   "source": [
+    "import numpy as np\n",
+    "\n",
+    "ceed = libceed.Ceed()\n",
+    "x = ceed.Vector(size=3)\n",
+    "\n",
+    "a = np.arange(1, 4, dtype=\"float64\")\n",
+    "x.set_array(a, cmode=libceed.USE_POINTER)\n",
+    "b = x.get_array_read()\n",
+    "x.restore_array_read()\n",
+    "\n",
+    "print(b)"
```

Comment:
This is a bug, and why I wanted to use a `with` context manager to prevent use-after-restore. We don't want `b` to be able to outlive `x.restore_array_read()`.

Reply:
Thanks for catching that! I thought I had fixed it already, but it must have slipped.